### ISSUE TITLE
Cater for subjects done by whole classes

### ISSUE THREAD

**Issue by avishkar58** [avishkar58/primate#19]
Heideveld offers Computers as a subject which all classes do. How do we represent this efficiently in the structure.

**Comment by avishkar58** [comment]
For now we will add Computers under the Subject node in the tree and assign each student who does computers to that level (as well as their class). When we initially brainstormed the structure, we figured it would be flexible enough to cater for all kinds of school structures and class/subject assignments, which I believe it is. We may find, however, that the flexibility makes it too difficult to work with or maintain, in which case we can use the (actual) feedback to refine the structure and/or change the way it works.

**Comment by SadiqNacerodien** [comment]
Can we (in that case) add a feature that adds, in 1 click, all students in a particular class or level?

For example, maybe only GR4a does Computers, so on that class have a button to add to the Subject Computers? Or, if they want to wider, all of the Foundation phase? Seems a bit complicated though...almost leaning towards having the concept of Subjects vs Classes (not as the branch of the same tree)...but lets play it by ear

**Comment by avishkar58** [comment]
Yeah that sounds like a reasonable compromise that will make maintenance easier for (primary) schools where entire classes do a subject...
